tests/unit_tests/test_optional_part.py

```diff
@@ -0,0 +1,191 @@
+    current_year = 2023 
+    item = Item(year=2013)
+    # (current_year - item.year (2023-2013) = 10)
+    expected = 10
+    
+    actual = item.calculate_age()
+    
+    assert actual == expected
+    assert isinstance(actual, int)
+    
+#@pytest.mark.skip     
+def test_calculate_age_returns_false_if_no_year_is_given():
+    item = Item()
+    
+    actual = item.calculate_age()
+    
+    assert actual is False
+    
+
+'''vendor.py add_ons'''
+
+#@pytest.mark.skip 
+def test_get_by_newest_returns_youngest_item_from_non_empty_inventory():
+    item_a = Item(year=2018)
+    item_b = Item(year=2006)
+    item_c = Item(year=1978)
+    inventory = [item_a, item_b, item_c]
+    
+    vendor = Vendor(inventory)
+    expected = item_a
+    
+    actual = vendor.get_by_newest()
+    
+    assert actual is item_a
+
+#@pytest.mark.skip     
+def test_get_by_newest_returns_false_if_empty_inventory():
+    inventory = []
+    vendor = Vendor(inventory)
+    
+    actual = vendor.get_by_newest()
+    
+    assert actual is None
+    
+#@pytest.mark.skip     
+def test_get_by_newest_returns_newest_if_some_items_has_year_with_default_value():
+    item_a = Item(year=2018)
+    item_b = Item()
+    item_c = Item(year=1978)
+    inventory = [item_a, item_b, item_c]
+    vendor = Vendor(inventory)
+    
+    expected = item_a
+    
+    actual = vendor.get_by_newest()
+    
+    assert actual == expected
+    assert len(inventory) == len(vendor.inventory)
+
+#@pytest.mark.skip    
+def test_swap_by_newest_swaps_self_newest_item_and_others_newest_item_succesfully():
+    item_a = Item(year=2018)
+    item_b = Item(year=2006)
+    item_c = Item(year=1978)
+    inventory = [item_a, item_b, item_c]
+    us = Vendor(inventory)
+    
+    item_x = Item(year=2003)
+    item_y = Item(year=2017)
+    item_z = Item(year=1953)
+    other_inventory = [item_x, item_y, item_z]
+    other_vendor = Vendor(other_inventory)
+    
+    our_expected_inventory = [item_b, item_c, item_y]
+    their_expected_inventory = [item_x, item_z, item_a]
+    
+    us.swap_by_newest(other_vendor)
+    
+    assert len(us.inventory) == 3
+    assert len(other_vendor.inventory) == 3
+    assert item_a not in us.inventory
+    assert item_y not in other_vendor.inventory
+    assert us.inventory == our_expected_inventory
+    assert other_vendor.inventory == their_expected_inventory
+    
+#@pytest.mark.skip
+def test_swap_by_newest_returns_false_if_ones_inventory_is_empty():
+    item_a = Item(year=2018)
+    item_b = Item(year=2006)
+    item_c = Item(year=1978)
+    inventory = [item_a, item_b, item_c]
+    us = Vendor(inventory)
+    
+    other_inventory = []
+    other_vendor = Vendor(other_inventory)
+    
+    result = us.swap_by_newest(other_vendor)
+    
+    assert len(us.inventory) == 3
+    assert len(other_vendor.inventory) == 0
+    assert us.inventory == inventory
+    assert other_vendor.inventory == []
+    assert not result
```

Comment:
[suggestion] when you want to test that `False`, the boolean, actually came out of the function, you might prefer `assert result is False`.